man/fi_domain.3.md

```diff
@@ -666,6 +667,78 @@ Applications can set the mr_cnt on input to fi_getinfo, in order to
+
+Traffic classes can be set according to two different types of classifications
+available to libfabric: FI_TC_TYPE_LABEL or FI_TC_TYPE_DSCP. If the traffic class
+*type* field is set to FI_TC_TYPE_DSCP, the value field of the traffic class must be set
+to a DSCP value. If the traffic class *type* field is set to FI_TC_TYPE_LABEL, the
+value field of the traffic class must be set to a libfabric or provider-specific
+defined label of a traffic class.
+
+Traffic classes can be any of the following defined classes of traffic,
+or any provider-specific definitions.
+
+** FI_TC_BEST_EFFORT **
+** FI_TC_LOW_LATENCY **
+** FI_TC_DEDICATED_ACCESS **
+** FI_TC_BULK_IO_DATA **
+** FI_TC_SCAVENGER **
+** FI_TC_MANAGEMENT **
+** FI_TC_BULK_IPC_DATA **
+
+The general behavior provided for each traffic class is provided below.
+
+*FI_TC_BEST_EFFORT*
+: This is the default, shared class, carrying the traffic for a number of
+applications executing concurrently over the same network infrastructure.
+Even though it is shared, network capacity and resource allocation are
+distributed fairly across the applications.
+
+*FI_TC_LOW_LATENCY*
+: This class supports low latency, low jitter data patterns typically caused by
+transactional data exchanges, barrier synchronizations, and collective operations
+that are typical of HPC applications. Specified maximum latencies are requested
+by this class.  Fulfillment of such requests in this class will typically require
+accompanying bandwidth caps and packet size limitations so as not to consume
+excessive bandwidth at high priority.
+
+*FI_TC_DEDICATED_ACCESS*
+: This class operates at the highest priority, except the management class.
+It carries a high bandwidth allocation, maximum latency targets, and the
+highest scheduling and arbitration priority.
+
+*FI_TC_BULK_IO_DATA*
+: This class is intended for large data transfers associated with I/O and
+is present to separate sustained I/O transfers from other application
+inter-process communications.
+
+*FI_TC_SCAVENGER*
+: This class is used for data that is desired but does not have strict delivery
+requirements such as monitoring data especially for application-specific
+monitoring such as with performance tools.  Global monitoring could also be done
+this way for data that is too voluminous for out-of-band transport.  Using this
+class makes sure that such communication does not interfere with communication doing “real” work.
+
+*FI_TC_BULK_IPC_DATA*
+: This class is intended to support large IPC transfers such as MPI rendezvous operations.
+
+*FI_TC_MANAGEMENT*
+: This class is intended for traffic directly related to fabric (network) management.
+It should not be used by standard user or kernel applications.
+
+Libfabric reserves the 32 least significant bits of the traffic class *value* field
+for predefined traffic class labels as provided above. Providers may implement
+provider-specific labels with the remaining 32 bits of the traffic class *value* field.
```

Comment:
The value field can be 32-bits total.  If the FI_PROV_SPECIFIC bit flag is set, then the value can be interpreted as provider specific.

Reply:
Sure, that could work.